build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
++      } else if (GetDeclarationValue(line, kNameDeclaration, &value)) {
++        script->set_name(value);
++      } else if (GetDeclarationValue(line, kVersionDeclaration, &value)) {
++        base::Version version(value);
++        if (version.IsValid())
++          script->set_version(version.GetString());
++      } else if (GetDeclarationValue(line, kDescriptionDeclaration, &value)) {
++        script->set_description(value);
++      } else if (GetDeclarationValue(line, kMatchDeclaration, &value)) {
++        URLPattern pattern(UserScript::ValidUserScriptSchemes());
++        if (URLPattern::ParseResult::kSuccess != pattern.Parse(value)) {
++          error_message = "Invalid UserScript Schema " + value;
++          return false;
++        }
++        script->add_url_pattern(pattern);
++      } else if (GetDeclarationValue(line, kExcludeMatchDeclaration, &value)) {
++        URLPattern exclude(UserScript::ValidUserScriptSchemes());
++        if (URLPattern::ParseResult::kSuccess != exclude.Parse(value)) {
++          error_message = "Invalid UserScript Schema " + value;
++          return false;
++        }
++        script->add_exclude_url_pattern(exclude);
++      } else if (GetDeclarationValue(line, kRunAtDeclaration, &value)) {
++        if (value == kRunAtDocumentStartValue)
++          script->set_run_location(UserScript::DOCUMENT_START);
++        else if (value == kRunAtDocumentEndValue)
++          script->set_run_location(UserScript::DOCUMENT_END);
++        else if (value == kRunAtDocumentIdleValue)
++          script->set_run_location(UserScript::DOCUMENT_IDLE);
++        else {
++          error_message = "Invalid RunAtDeclaration " + value;
++          return false;
++        }
++      } else if(GetDeclarationValue(line, kUrlSourceDeclaration, &value)) {
++        script->set_url_source(value);
++      }
++
++      // TODO(aa): Handle more types of metadata.
++    }
++
++    line_start = line_end + 1;
++  }
++
++  // If no patterns were specified, default to @include *. This is what
++  // Greasemonkey does.
++  if (script->globs().empty() && script->url_patterns().is_empty())
++    script->add_glob("*");
++
++  return true;
++}
++
++// static
++bool LoadUserScriptFromFile(
++    const base::FilePath& user_script_path, const GURL& original_url,
++    std::unique_ptr<UserScript>& script,
++    base::string16* error) {
++
++    std::string script_key = user_script_path.BaseName().value();
++
++    std::string content;
++    if (user_script_path.IsContentUri()) {
++      if (base::FeatureList::IsEnabled(features::kEnableLoggingUserScripts))
++        LOG(INFO) << "UserScriptLoader: path " << user_script_path << " is a content uri";
++
++      base::FilePath tempFilePath;
++      if( base::CreateTemporaryFile(&tempFilePath) == false ) {
++        *error = base::ASCIIToUTF16("Could not read create temp file.");
++        return false;
++      }
++
++      if( base::CopyFile(user_script_path, tempFilePath) == false ) {
```

Comment:
```suggestion
+      if (base::CopyFile(user_script_path, tempFilePath) == false) {
```

Reply:
fixed in https://github.com/bromite/bromite/pull/857/commits/8c164138391082350f20493a0ca5c4926887372c

completely removed the copy in the temps